### PR TITLE
Redesign comments

### LINKED ISSUE #63: Redesign comments
### Summary
- I want to see count of comments
- I want to see when a comment was left
- The name of the comment's author should be more visible than it is now

### References
image
- https://dribbble.com/shots/17196663-Comments
- https://dribbble.com/shots/17464378-Task-and-Comment

### PR DESCRIPTION
**Pull request by RasimKhusaenov**
### Summary

- Comments sector redesigned
- Feature test was refactored, because there is no label now

### How it works

image
image

### Test plan

List of steps to manually test introduced functionality:

- Log in
- Go to https://my-own-blog-pr-64.herokuapp.com/articles/2
- Make sure that comments displayed correctly and there is no scroll when your comment text is too big.

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, ENV variables, services, scripts, etc.

### Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests for new functionality
- [x] My pull request is shorter than 500 lines

### References
- Resolves #63 Issue
- GitHub Pull Request #62

### CODE REVIEW

File: app/views/comments/_list.html.slim [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
-  ul class="list-unstyled"
-    - comments.each do |comment|
-      li class="py-1"
+  - comments.each do |comment|
+    div class="flex-column border-bottom border-1 mb-1 pb-1"
+      div class="d-flex align-items-center"
         = comment.user.first_name
-        ' :
+        '  ·
+        a class="text-muted ps-1 text-decoration-none comment__date"
```
Comment (VeraKutyavina) [inline app/views/comments/_list.html.slim:7]: better to use `span`

File: app/views/articles/show.html.slim [hunk 2]
```diff
@@ -6,9 +6,12 @@ p #{article.content}
...
-  p
-    b Comments:
-    - if allowed_to?(:create?, comment, with: Users::CommentPolicy)
-      = render partial: "comments/form"
-    = render partial: "comments/list"
+section class="border-top border-1 pt-2"
+  h5 class="d-flex align-items-center"
+    | Comments
+    - if comments.size.positive?
+      span class="badge bg-secondary rounded-pill ms-1 p-1"
```
Comment (VeraKutyavina) [inline app/views/articles/show.html.slim:13]: it's look like `div`

File: app/views/comments/_list.html.slim [hunk 3]
```diff
@@ -1,9 +1,14 @@
...
+          = time_ago_in_words comment.created_at
+          '  ago
+      p class="text-secondary mb-0"
```
Comment (VeraKutyavina) [inline app/views/comments/_list.html.slim:10]: you can add  `white-space: pre-line;` to save user comment formatting

**Review by RasimKhusaenov — COMMENTED** [review]
There is an old scss in review app that doesn't want to be recompiled. I can't fix it right now. :smiling_face_with_tear:

File: app/assets/stylesheets/form.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
+.comment__content {
+  white-space: pre-line;
+}
+
+.comment__counter {
+  min-width: 1.3rem;
+  height: 1.3rem;
+  font-size: 0.8rem;
+  padding-right: .3em;
+  padding-left: .3em;
```
Comment (VeraKutyavina) [inline app/assets/stylesheets/form.scss:10]: you can use `padding: 0 .3em`

File: app/views/comments/_list.html.slim [hunk 5]
```diff
@@ -1,9 +1,14 @@
...
-    - comments.each do |comment|
-      li class="py-1"
-        = comment.user.first_name
-        ' :
+  - comments.each do |comment|
+    div class="flex-column border-bottom border-1 mb-1 pb-1"
+      div class="d-flex align-items-center"
+        => comment.user.first_name
+        | ·
+        span class="text-muted ps-1 text-decoration-none comment__date"
```
Comment (VeraKutyavina) [inline app/views/comments/_list.html.slim:7]: Do not need text-decoration-none class, there are no special styles for span

File: app/views/comments/_list.html.slim [hunk 6]
```diff
@@ -1,9 +1,14 @@
 - if comments.present?
-  ul class="list-unstyled"
-    - comments.each do |comment|
-      li class="py-1"
-        = comment.user.first_name
-        ' :
+  - comments.each do |comment|
+    div class="flex-column border-bottom border-1 mb-1 pb-1"
```
Comment (VeraKutyavina) [inline app/views/comments/_list.html.slim:3]: i think mb-2 pb-2 looks better, just my opinion

**Review by VeraKutyavina — APPROVED** [review]

**Pull request merged**